buildconf

```diff
@@ -90,6 +90,25 @@ if test ! -f configure.ac ||
   exit 1
 fi
 
+
+
+
+#--------------------------------------------------------------------------
+# Ensure that brew is installed and install if not
+#
+# am4te_version=`${AUTOM4TE:-autom4te} --version 2>/dev/null|head -n 1| sed -e 's/autom4te\(.*\)/\1/' -e 's/^[^0-9]*//' -e 's/[a-z]* *$//'`
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  brew=`${BREW:-brew} -v 2>/dev/null|head -n 1| sed -e 's/Homebrew^[^0-9]*//' -e 's/[a-z]* *$//'`
+  if test -z "$brew"; then
+    echo "Installing Homebrew ..."
+    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
```

Comment:
I'm really not convinced that `buildconf` should have these side-effects.

Reply:
why not, I mean we could have both , the documentation for Homebrew and we could automatically install them  ?

Reply:
because that's not how we typically use that script and I don't think users have given the script that permission. We don't know that people want those tools installed, we don't know they want them installed with homebrew, we don't know if perhaps only the PATH was wrong etc.

We could certainly output text that offers help on exactly what and how to install the missing pieces.

Reply:
during curl's lifetime there has been many package managers and no doubt there will be more so less enthusiastic about side effects from buildconf.

+1 to 'output text' suggestion ... that seems safe enough ... granted this is probably only useful for new developers of curl in the first # mins.

Reply:
Ok , it seems every body not convinced and i am as well convencid about the side effects

So the question where to add documentation instead?

Reply:
Or i could work on https://github.com/curl/curl/issues/3229

Reply:
maybe add to this area of the site (or add a top level 'install dependencies') ?
https://curl.haxx.se/dev/internals.html#Portability

Reply:
Isn't there basically a single command line to install of them at once, assuming homebrew is already installed? That seems like an excellent thing to add to `docs/INSTALL.md`. It'd be neat to have for more distros/OSes.

Reply:
Yes you can in 1 line but the point is could be the user have one from two of them

Ok i this i will add the documentation in docs/INSTALL.md and what about add the current starus in buildconf as well but with a question 
Do you want to install missing depend. With Homebrew y/n ?

Reply:
> but the point is could be the user have one from two of them

Ok sure, but then that just becomes a no-op for that component doesn't it?

> what about add the current starus in buildconf as well but with a question
> Do you want to install missing depend. With Homebrew y/n ?

Sure, I'm willing to give that a try!

Reply:
I have done an update